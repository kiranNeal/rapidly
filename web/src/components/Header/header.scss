@import "../../constants/styleVariables.scss";

.header__container {
    width: 100%;
    font-size: 18px;
    background-color: $darkGrey;
    color: $white;
    display: flex;
    flex-direction: row;
    align-items: center;
    align-self: center;
    padding: 0.4rem 0;
    z-index: 666;
    justify-content: flex-start;
}

.header__icon {
    height: 30px;
    margin-left: 10px;
}

.header__title {
    width: 100%;
    display: flex;
    justify-content: flex-start;
    font-weight: bold;
    margin-left: 1%;
}
