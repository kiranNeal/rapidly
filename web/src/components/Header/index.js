import React from 'react';

import './header.scss';
import Logo from '../../assets/icons/rapidlyIcon.png';

export default function Header() {
    return (
      <header className="header__container">
        <img className="header__icon" src={Logo} alt="rapidly" />
        <div className="header__title">RAPIDLY</div>
      </header>
    );
}
