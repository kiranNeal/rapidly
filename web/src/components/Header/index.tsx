import React from 'react';

import './header.scss';
import Logo from '../../assets/icons/rapidlyIcon.png';

const Header: React.FC = () => {
  return (
    <header className="header__container">
      <img className="header__icon" src={Logo} alt="rapidly" />
      <div className="header__title">RAPIDLY</div>
    </header>
  );
};

export default Header;
