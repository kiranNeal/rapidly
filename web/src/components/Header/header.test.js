import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Header from './index.js';

describe('Header component tests', () => {
    afterEach(cleanup);

    it('Header renders as expected', () => {
        const { asFragment } = render(<Header />);
        expect(asFragment()).toMatchSnapshot();
    });

    it('Header renders Title correctly', () => {
        const { getByText } = render(<Header />);
        const linkElement = getByText('RAPIDLY');
        expect(linkElement).toBeInTheDocument();
    });
});



