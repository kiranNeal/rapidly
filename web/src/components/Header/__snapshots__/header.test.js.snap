// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests Header renders 1`] = `
<DocumentFragment>
  <header
    class="header__container"
  >
    <img
      alt="rapidly"
      class="header__icon"
      src="rapidlyIcon.png"
    />
    <div
      class="header__title"
    >
      RAPIDLY
    </div>
  </header>
</DocumentFragment>
`;
